main.c

```diff
@@ -84,6 +73,16 @@ int main()
     /* Initialize UART. */
     uart_init();
     uart_putstr("Hello, UART!\n");
+
+    /* Demonstrate access to .data */
+    uart_putstr(str);
+
+    /* Test whether .bss appears to have been cleared. */
+    char* p = empty;
+    while(p < empty + sizeof(empty))
```

Comment:
I can see code formatting inconsistencies, including whitespace mix of tabs and spaces. That isn't worrying at the moment but it'll become a problem in long run. Perhaps we could use GNU indent to solve the problem?